### PR TITLE
Remove legacy JSON code

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This PR fixes up @ngrodzitski's work in #1343 so we can merge it into master.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

**Review by tobim — APPROVED** [review]
All good as far as I can tell.
**edit:** I did test the output of `vast version` and `vast status`. The other stuff should be covered by CI.

**Comment by dominiklohmann** [comment]
> I did test the output of `vast version` and `vast status`. The other stuff should be covered by CI.

They are both covered by integration tests. The example plugin integration test checks `vast version` , and `vast status` we cover in multiple places.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,20 @@ This changelog documents all notable user-facing changes of VAST.
 
 ## Unreleased
 
+### ⚡️ Breaking Changes
```
Comment (tobim) [inline CHANGELOG.md:16]: Uh, was this added intentionally?

Comment (dominiklohmann) [reply CHANGELOG.md:16]: Nope, that was definitely a mistake. Can you remove this in your other PR that's likely next to be merged?

Comment (tobim) [reply CHANGELOG.md:16]: The one for the partition_selector is probably next, but that doesn't touch the changelog. I'll remove it in the PR that moves around the options.

Comment (dominiklohmann) [reply CHANGELOG.md:16]: Thanks!

**Pull request merged**